pkg/account/account.go

```diff
@@ -0,0 +1,339 @@
+}
+
+// Update the Account record in DynamoDB
+func (a *Account) Update(d model.Account, am Manager) error {
+	err := validation.ValidateStruct(&d,
+		// ID has to be empty
+		validation.Field(&d.ID, validation.NilOrNotEmpty, validation.In(*a.data.ID)),
+		validation.Field(&d.AdminRoleArn, validation.By(isNilOrUsableAdminRole(am))),
+		//validation.Field(&d.AdminRoleArn, validation.By(isNil)),
+		validation.Field(&d.ID, validation.By(isNil)),
+		validation.Field(&d.LastModifiedOn, validation.By(isNil)),
+		validation.Field(&d.Status, validation.By(isNil)),
+		validation.Field(&d.CreatedOn, validation.By(isNil)),
+		validation.Field(&d.PrincipalRoleArn, validation.By(isNil)),
+		validation.Field(&d.PrincipalPolicyHash, validation.By(isNil)),
+	)
+	if err != nil {
+		return errors.NewValidation("account", err)
+	}
+
+	if d.AdminRoleArn != nil {
+		a.data.AdminRoleArn = d.AdminRoleArn
+	}
+	if d.Metadata != nil {
+		a.data.Metadata = d.Metadata
+	}
+
+	err = a.save()
+	if err != nil {
+		return fmt.Errorf("unable to update account %s: %w", *a.data.ID, err)
+	}
+	return nil
+}
+
+// Delete finds a given account and deletes it if it is not of status `Leased`. Returns the account.
+func (a *Account) Delete() error {
+
+	err := validation.ValidateStruct(&a.data,
+		validation.Field(&a.data.Status, validation.NotNil, validation.By(isAccountNotLeased)),
+	)
+	if err != nil {
+		return errors.NewConflict("account", *a.data.ID, err)
+	}
+
+	err = a.writer.DeleteAccount(&a.data)
+	if err != nil {
+		return fmt.Errorf("unable to delete account \"%s\": %w", *a.data.ID, err)
+	}
+
+	return nil
+}
+
+// OrphanAccount - Orpahn an account
+func (a *Account) OrphanAccount() error {
+	return nil
+}
+
+// GetAccountByID returns an account from ID
+func GetAccountByID(ID string, d SingleReader, wd WriterDeleter) (*Account, error) {
+
+	newAccount := Account{
+		writer: wd,
+	}
+	err := d.GetAccountByID(ID, &newAccount.data)
+
+	return &newAccount, err
+}
+
+// New returns an account from ID
+func New(wd WriterDeleter, data model.Account) *Account {
+	now := time.Now().Unix()
+	account := &Account{
+		writer: wd,
+		data:   data,
+	}
+	account.data.CreatedOn = &now
+	account.data.LastModifiedOn = &now
+	return account
+}
+
+// MarshalJSON Marshals the data inside the account
+func (a *Account) MarshalJSON() ([]byte, error) {
+	return json.Marshal(a.data)
+}
+
+// GetReadyAccount returns an available account record with a
+// corresponding status of 'Ready'
+func GetReadyAccount(d Reader, wd WriterDeleter) (*Account, error) {
+	accounts, err := GetAccountsByStatus(model.AccountStatus("Ready"), d)
+	if err != nil {
+		return nil, err
+	}
+	if len(*accounts) < 1 {
+		return nil, errors.NewNotFound("account", "ready")
+	}
+
+	account := &(*accounts)[0]
+	account.writer = wd
+
+	return account, err
+}
+
+// CreateAccount creates a new account
+func CreateAccount(ID string, AdminRoleArn string, Metadata map[string]interface{}, r Reader, wd WriterDeleter) (*Account, error) {
```

Comment:
This function isn't used currently.  Only here to stub it out as it will be used at some point but more work is needed.  Eventer service needs to get put in and the account manager service needs to get put in.